app/services/request_create_service.rb

```diff
@@ -88,9 +91,18 @@ def request_finished(request)
   end
 
   def start_first_stage(request)
+    request_started(request)
     ActionCreateService.new(request.stages.first.id).create(
       :operation    => Action::NOTIFY_OPERATION,
       :processed_by => 'system'
     )
   end
+
+  def start_external_approval_process(request)
+    template = request.workflow.template
+    processor_class = "#{template.process_setting['processor_type']}_process_service".classify.constantize
```

Comment:
What's the benefit we use such a way here and next to initialize `jbpm_process_service`? This will require both `process_setting `and `signal_setting` (in `stage_update_service`) to have attribute `processor_type`. What happens if it's not been defined?

Reply:
We talked about supporting of external approval processors. The `processor_type` allows us to expand such capacity.

Reply:
👍

Reply:
Not for this PR, but would be good to move this line for figuring out the `process_class` into a separate method at some point.  Would help with testing when we do support external processors.